databricks/koalas/window.py

```diff
@@ -42,6 +60,369 @@ def __getattr__(self, item: str) -> Any:
+        >>> df = ks.DataFrame({"A": s.to_numpy(), "B": s.to_numpy() ** 2})
+        >>> df
+           A   B
+        0  4  16
+        1  3   9
+        2  5  25
+        3  2   4
+        4  6  36
+
+        >>> df.rolling(2).max()
+             A     B
+        0  NaN   NaN
+        1  4.0  16.0
+        2  5.0  25.0
+        3  5.0  25.0
+        4  6.0  36.0
+
+        >>> df.rolling(3).max()
+             A     B
+        0  NaN   NaN
+        1  NaN   NaN
+        2  5.0  25.0
+        3  5.0  25.0
+        4  6.0  36.0
+        """
+        def max(scol, index_scols):
+            window = Window.orderBy(index_scols)
+            return F.when(
+                F.lag(scol, self._window_val).over(window) >= self._min_periods,
+                F.max(scol).over(self._window)
+            ).otherwise(F.lit(None))
+
+        return self._apply_as_series_or_frame(max)
+
+    def mean(self):
+        """
+        Calculate the rolling mean of the values.
+
+        .. note:: the current implementation of this API uses Spark's Window without
+            specifying partition specification. This leads to move all data into
+            single partition in single machine and could cause serious
+            performance degradation. Avoid this method against very large dataset.
+
+        Returns
+        -------
+        Series or DataFrame
+            Returned object type is determined by the caller of the rolling
+            calculation.
+
+        See Also
+        --------
+        Series.rolling : Calling object with Series data.
+        DataFrame.rolling : Calling object with DataFrames.
+        Series.mean : Equivalent method for Series.
+        DataFrame.mean : Equivalent method for DataFrame.
+
+        Examples
+        --------
+        >>> s = ks.Series([4, 3, 5, 2, 6])
+        >>> s
+        0    4
+        1    3
+        2    5
+        3    2
+        4    6
+        Name: 0, dtype: int64
+
+        >>> s.rolling(2).mean()
+        0    NaN
+        1    3.5
+        2    4.0
+        3    3.5
+        4    4.0
+        Name: 0, dtype: float64
+
+        >>> s.rolling(3).mean()
+        0         NaN
+        1         NaN
+        2    4.000000
+        3    3.333333
+        4    4.333333
+        Name: 0, dtype: float64
+
+        For DataFrame, each rolling max is computed column-wise.
+
+        >>> df = ks.DataFrame({"A": s.to_numpy(), "B": s.to_numpy() ** 2})
+        >>> df
+           A   B
+        0  4  16
+        1  3   9
+        2  5  25
+        3  2   4
+        4  6  36
+
+        >>> df.rolling(2).mean()
+             A     B
+        0  NaN   NaN
+        1  3.5  12.5
+        2  4.0  17.0
+        3  3.5  14.5
+        4  4.0  20.0
+
+        >>> df.rolling(3).mean()
+                  A          B
+        0       NaN        NaN
+        1       NaN        NaN
+        2  4.000000  16.666667
+        3  3.333333  12.666667
+        4  4.333333  21.666667
+        """
+        def mean(scol, index_scols):
```

Comment:
Why do we need to send `index_scols`? Seems it can be set within `_apply_as_series_or_frame`.

Reply:
ah, you're right, i'll fix them and also https://github.com/databricks/koalas/pull/990#discussion_r342359478

Reply:
Thanks for the comment! i fixed them